internal/stest/assertions.go

```diff
@@ -124,10 +124,93 @@ func AssertPartialMatch(t *testing.T, evs []cap.Event, preds []EventP) {
 	}
 }
 
+// ObserveDynSupervisor is an utility function that receives all the arguments
+// required to build a DynSupervisor, and a callback that when executed will
+// block until some point in the future (after we performed the side-effects we
+// are testing). This function returns the list of events that happened in the monitored
+// supervised tree, as well as any crash errors.
+func ObserveDynSupervisor(
+	ctx context.Context,
+	rootName string,
+	childNodes []cap.Node,
+	opts0 []cap.Opt,
+	callback func(cap.DynSupervisor, EventManager),
+) ([]cap.Event, []error) {
+	evManager := NewEventManager()
+	// Accumulate the events as they happen
+	evManager.StartCollector(ctx)
+
+	// Create a new Supervisor Opts that adds the EventManager's Notifier at the
+	// very beginning of the system setup, the order here is important as it
+	// propagates to sub-trees specified in this options
+	opts := append([]cap.Opt{
+		cap.WithNotifier(evManager.EventCollector(ctx)),
+	}, opts0...)
+
+	// We always want to start the supervisor for test purposes, so this is
+	// embedded in the ObserveDynSupervisor call
+	sup, err := cap.NewDynSupervisor(ctx, rootName, opts...)
+
+	if err != nil {
+		return evManager.Snapshot(), []error{err}
+	}
+
+	errors := []error{}
+
+	// start procedurally the children given children
+	for _, node := range childNodes {
+		_, spawnErr := sup.Spawn(node)
+		if spawnErr != nil {
+			errors = append(errors, spawnErr)
+		}
+	}
+
+	evIt := evManager.Iterator()
+
+	if len(errors) != 0 {
+		// once tests are done, we stop the supervisor
+		if err = sup.Terminate(); err != nil {
+			errors = append(errors, err)
+		}
+		evIt.SkipTill(SupervisorTerminated(rootName))
+		return evManager.Snapshot(), errors
+	}
+
+	// NOTE: We execute SkipTill to make sure all the supervision tree got started
+	// (or failed) before doing assertions/returning an error. Also, note we use
+	// ProcessName instead of ProcessStarted/ProcessFailed given that ProcessName
+	// matches an event in both success and error cases. The event from root must
+	// be the last event reported
+	evIt.SkipTill(ProcessName(rootName))
+
+	if err != nil {
```

Comment:
I am lost. Where did this `err` come from? As best as I can tell. last time you set it and checked it, you returned?